index.js

```diff
@@ -21,16 +36,29 @@ module.exports = function (str, opts) {
 			}
 		}
 
-		// start /* type comment
+		//find beginning of /* type comment
 		if (!insideString && currentChar === '/' && str[i + 1] === '*') {
-			if (!preserve || preserve && str[i + 2] !== '!') {
-				// start skipping until we reach end of comment
+			//ignore important comment when configured to preserve comments using important syntax: /*!
+			if (!(preserveImportant && str[i + 2] === '!')) {
+				// iterate over comment
 				for (var j = i + 2; j < str.length; j++) {
+					//find end of comment
 					if (str[j] === '*' && str[j + 1] === '/') {
+						if (preserveFilter) {
+							//evaluate comment text
+							ret = preserveFilter(comment) ? ret += ('/*' + comment + '*/') : ret;
```

Comment:
True ternary result should just be `ret + ...` instead of `ret += ...` because you're assigning to `ret` anyway.